### ISSUE TITLE
[docs] Remove column id anywhere its not displayed

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-x#346]
Based on #342, we can simplify the demos of the documentation to not include the ID column if displayed hidden.

```diff
const columns: ColDef[] = [
- { field: 'id', hidden: true },
  { field: 'firstName', headerName: 'First name', width: 130 },
  { field: 'lastName', headerName: 'Last name', width: 130 },
```

For instance on the https://material-ui.com/components/data-grid/rendering/ page:

https://github.com/mui-org/material-ui-x/blob/6d23e1afb344d783727c484a41b5a8e1fd2cb5e1/docs/src/pages/components/data-grid/rendering/ValueGetterGrid.tsx#L5

**Comment by kodeschooler** [comment]
Not sure if this is the same issue, but I have my ID field configured like this: 

```
{
            field: "Id",
            hide: true,
            hideable: false,
            filterable: false,
          }
```

And when you go to "Show Columns" (on any column header) it displays the ID in the Show All / Hide All menu with the switch disabled. But how do we not show it at all? The ID is required by the grid so we had to use it, but we don't want it to be displayed anywhere as it has no meaning to a user. Can this be done?

**Comment by m4theushw** [comment]
@kodeschooler Could you open another issue? This one was only to hide the ID column in our demos. Regarding not displaying `hideable=false` columns in the filter painel, I asked about that in https://github.com/mui-org/material-ui-x/pull/3433#discussion_r768871395. Maybe we need to add another option to configure the behavior.

**Comment by kodeschooler** [comment]
Ah. I will, sure, but now I'm a little more confused. I thought an ID column was required for the grid. Then I looked at the demos where you guys have an ID in the row data but don't have an ID column defined in the columns. So I just commented out my ID column and it doesn't throw an error about a missing ID. And, it actually produces the result I want- which is to not have ID show in the hide/show menu (because I'm not displaying it). 

Is ID only required in the rows data, but not in the columns? Or is it not required in either? Maybe I'm confusing it with something else and ID is not in fact required.  If that's the case I don't really have an issue.  I thought at one point recently I got an error about it missing.  

As far as not displaying hideable=false columns in filter panel, they don't. I just checked.
